core/src/main/java/com/linecorp/armeria/server/http/auth/HttpAuthServiceBuilder.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.auth;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * Builds a new {@link HttpAuthService}.
+ */
+public final class HttpAuthServiceBuilder {
+
+    private Service<? super HttpRequest, ? extends HttpResponse> delegate;
+    private final List<Predicate<? super HttpHeaders>> predicates = new ArrayList<>();
+
+    /**
+     * Sets a delegating service.
+     *
+     * @throws IllegalArgumentException if the delegating service is already set.
+     */
+    public HttpAuthServiceBuilder of(Service<? super HttpRequest, ? extends HttpResponse> delegate) {
```

Comment:
Other builders actually require the delegate in `build()` method. Could you make this builder has the same style of API? See `CorsServiceBuilder.build()` and `newDecorator()` for an example.